onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -458,72 +446,66 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
+              "'GPU.0_FP16', 'GPU.1_FP16' or from"
               " HETERO/MULTI/AUTO options available. \n");
         }
       } else if (key == "device_id") {
-        device_id = value;
+        ov_options[key] = value;
       } else if (key == "enable_vpu_fast_compile") {
-        if (value == "true" || value == "True") {
-          enable_vpu_fast_compile = true;
-        } else if (value == "false" || value == "False") {
-          enable_vpu_fast_compile = false;
+        if (value == "true" || value == "True" ||
+            value == "false" || value == "False") {
+          ov_options[key] = value;
         } else {
           ORT_THROW("[ERROR] [OpenVINO] The value for the key 'enable_vpu_fast_compile' should be a boolean i.e. true or false. Default value is false.\n");
         }
       } else if (key == "enable_opencl_throttling") {
-        if (value == "true" || value == "True") {
-          enable_opencl_throttling = true;
-        } else if (value == "false" || value == "False") {
-          enable_opencl_throttling = false;
+        if (value == "true" || value == "True" ||
+            value == "false" || value == "False") {
+          ov_options[key] = value;
         } else {
           ORT_THROW("[ERROR] [OpenVINO] The value for the key 'enable_opencl_throttling' should be a boolean i.e. true or false. Default value is false.\n");
         }
       } else if (key == "enable_dynamic_shapes") {
-        if (value == "true" || value == "True") {
-          enable_dynamic_shapes = true;
-        } else if (value == "false" || value == "False") {
-          enable_dynamic_shapes = false;
+        if (value == "true" || value == "True" ||
+            value == "false" || value == "False") {
+          ov_options[key] = value;
         } else {
           ORT_THROW(
               "[ERROR] [OpenVINO] The value for the key 'enable_dynamic_shapes' "
               "should be a boolean i.e. true or false. Default value is false.\n");
         }
       } else if (key == "num_of_threads") {
-        std::stringstream sstream(value);
-        sstream >> num_of_threads;
-        if ((int)num_of_threads <= 0) {
+        if (std::stoi(value) <= 0) {
           ORT_THROW("[ERROR] [OpenVINO] The value for the key 'num_of_threads' should be greater than 0\n");
+        } else {
+          ov_options[key] = value;
         }
       } else if (key == "cache_dir") {
-        cache_dir = value;
-      } else {
-        ORT_THROW("[ERROR] [OpenVINO] wrong key type entered. Choose from the following runtime key options that are available for OpenVINO. ['device_type', 'device_id', 'enable_vpu_fast_compile', 'num_of_threads', 'cache_dir', 'enable_opencl_throttling|true'] \n");
+        ov_options[key] = value;
+      } else if (key == "num_streams") {
+        if (std::stoi(value) <= 0 && std::stoi(value) > 8) {
+          ORT_THROW("[ERROR] [OpenVINO] The value for the key 'num_streams' should be in the range of 1-8 \n");
```

Comment:
I think in future if we add more features then this error wont hold. We should have more generic error statement